contentcuration/contentcuration/models.py

```diff
@@ -551,6 +552,78 @@ def get_thumbnail(self):
 
         return '/static/img/kolibri_placeholder.png'
 
+    def get_date_modified(self):
+        return self.main_tree.get_descendants(include_self=True).aggregate(last_modified=Max('modified'))['last_modified']
+
+    def get_resource_count(self):
+        return self.main_tree.get_descendants().exclude(kind_id=content_kinds.TOPIC).count()
+
+    def get_human_token(self):
+        return self.secret_tokens.get(is_primary=True)
+
+    def get_channel_id_token(self):
+        return self.secret_tokens.get(token=self.id)
+
+    def make_token(self):
+        """
+        Creates a primary secret token for the current channel using a proquint
+        string. Creates a secondary token containing the channel id.
+
+        These tokens can be used to refer to the channel to download its content
+        database.
+        """
+        token = proquint.generate()
+
+        # Try to generate the channel token, avoiding any infinite loops if possible
+        max_retries = 1000000
+        index = 0
+        while SecretToken.objects.filter(token=token).exists():
+            token = proquint.generate()
+            if index > max_retries:
```

Comment:
Where does index get incremented?

Reply:
Refactored it to use a for loop instead! Now we loop only 100 times and leave the incrementing to the for loop.